Riot/AppDelegate.m

```diff
@@ -1692,6 +1917,41 @@ - (nullable NSString *)limitedNotificationBodyForEvent:(NSString *)eventId inMat
     return notificationBody;
 }
 
+/**
+ Return the already displayed notification for an event.
+
+ @param eventId the id of the event attached to the notification to find.
+ @param userId the id of the user attached to the notification to find.
+ @param type the type of notification. @"full" or @"limited". nil for any type.
+ @return the local notification request if any.
+ */
+// iOS 10+ only!
+- (UNNotificationRequest *)displayedNotificationRequestForEvent:(NSString *)eventId andUser:(NSString *)userId type:(NSString*)type
```

Comment:
This method will always return nil. In fact `foundRequest` value will be returned at the end of the function before 
`getDeliveredNotificationsWithCompletionHandler:` completes, because this method is asynchronous.
But as mentioned above we could remove this method and rely on `UNNotificationRequest` identifier to avoid notifications duplication.

Reply:
https://github.com/vector-im/riot-ios/pull/2207/commits/90188cdcf36e82116580f0ea8fe1bd3367a0f089